analyzers/tests/SonarAnalyzer.UnitTest/Rules/ExpressionComplexityTest.cs

```diff
@@ -34,6 +34,66 @@ public class ExpressionComplexityTest
                 .WithOptions(ParseOptionsHelper.FromCSharp8)
                 .Verify();
 
+        [DataTestMethod]
+        [DataRow("==")]
+        [DataRow("!=")]
+        [DataRow("<")]
+        [DataRow("<=")]
+        [DataRow(">")]
+        [DataRow(">=")]
+        public void ExpressionComplexity_TransparentComparissionOperators(string @operator) =>
+            builderCS.AddSnippet($$$"""
+            class C
+            {
+                public void M()
+                {
+                    var x = true && true && (1 {{{@operator}}} (true ? 1 : 1));         // Compliant (Make sure, the @operator is not increasing complexity)
+                    var y = true && true && true && (1 {{{@operator}}} (true ? 1 : 1)); // Noncompliant {{Reduce the number of conditional operators (4) used in the expression (maximum allowed 3).}}
+                }
+            }
+            """)
+            .Verify();
+
+        [DataTestMethod]
+        [DataRow("o", "??")]
+        [DataRow("i", "|")]
+        [DataRow("i", "^")]
+        [DataRow("i", "&")]
+        [DataRow("i", ">>")]
+        [DataRow("i", ">>>")]
+        [DataRow("i", "<<")]
+        [DataRow("i", "+")]
+        [DataRow("i", "-")]
+        [DataRow("i", "*")]
+        [DataRow("i", "/")]
+        [DataRow("i", "%")]
+        public void ExpressionComplexity_TransparentBinaryOperators(string parameter, string @operator) =>
```

Comment:
I think you can add the UnsignedRightShifExpression test to the .CSharp11 file and build this with CSharp 8. Otherwise this whole test method should be inside preprocessor #if NET if I'm not mistaken.

Reply:
> I think you can add the UnsignedRightShifExpression test to the .CSharp11 file and build this with CSharp 8

I'd like to keep the test cases together.

> Otherwise this whole test method should be inside preprocessor #if NET if I'm not mistaken.

Partially true. You can use C#11 with Net48. It is not a supported scenario but something we do in our code base too. There is not much risk in running it on Net48 in this case.